core/control-plane/transfer-core/src/test/java/org/eclipse/edc/connector/transfer/transfer/TransferProcessManagerImplTest.java

```diff
@@ -231,6 +232,21 @@ void initial_manifestEvaluationFailed_shouldTransitionToTerminating() {
         });
     }
 
+    @Test
+    void initial_noPolicyFound_shouldTransitionToTerminating() {
+        when(transferProcessStore.nextForState(eq(INITIAL.code()), anyInt()))
```

Comment:
please add the explicit method stub `when(policyArchive.findPolicyForContract(any())).thenReturn(null)`

Reply:
Added the null check with https://github.com/eclipse-edc/Connector/pull/2816/commits/a3b3562c84f48f57a1f0f522baefd3a063cb2081